gm4_soul_fragments/data/gm4_soul_fragments/functions/soul_fragment/process.mcfunction

```diff
@@ -0,0 +1,31 @@
+# @s = abandoned soul fragment with item or player nearby
+# at @s
+# from soul_fragment/process_check
+
+# set scoreboard ID
+scoreboard players operation #current gm4_sf_id = @s gm4_sf_id
+
+# die if nearby linked player is at full hearts
+execute if score @p[gamemode=!creative,gamemode=!spectator,distance=..22,scores={gm4_sf_fragments=0}] gm4_sf_id = #current gm4_sf_id run stopsound @a[distance=..22] neutral minecraft:block.beacon.ambient
+execute if score @p[gamemode=!creative,gamemode=!spectator,distance=..22,scores={gm4_sf_fragments=0}] gm4_sf_id = #current gm4_sf_id run kill @s
+
+# visuals
+
+# emit particles and hum if revealed
+execute as @s[scores={gm4_sf_reveal=1..}] run function gm4_soul_fragments:soul_fragment/appear
+
+# check if there are soul revealing blocks nearby
+execute store result score @s gm4_sf_reveal run clone ~-3 ~-2 ~-3 ~3 ~2 ~3 ~-3 ~-2 ~-3 filtered #gm4_soul_fragments:soul_revealing force
```

Comment:
shouldn't this line go before line 15? Since this is the command that sets the score that line 15 checks